elephant/change_point_detection.py

```diff
@@ -0,0 +1,514 @@
+between a spike lying in the right and left window. If at any time `t` this 
+difference is large 'enough' is assumed the presence of a rate Change Point in 
+a neighborhood of `t`. A threshold `test_quantile` for the maximum of 
+the filter_process (max difference of spike count between the left and right 
+window) is derived based on asymptotic considerations. The procedure is repeated 
+for an arbitrary set of windows, with different size `h`.
+
+
+Example
+-------
+The following applies multiple_filter_test to a spike trains. 
+
+    >>> import quantities as pq
+    >>> import neo
+    >>> from elephant.multiple_filter_test import multiple_filter_test
+
+    
+    >>> test_array = [1.1,1.2,1.4,   1.6,1.7,1.75,1.8,1.85,1.9,1.95]
+    >>> st = neo.SpikeTrain(test_array, units='s', t_stop = 2.1)
+    >>> window_size = [0.5]*pq.s
+    >>> t_fin = 2.1*pq.s
+    >>> alpha = 5
+    >>> num_surrogates = 10000
+    >>> change_points = multiple_filter_test(window_size, st, t_fin, alpha,
+                        num_surrogates, dt = 0.5*pq.s)
+
+
+
+Reference
+---------
+Messer, M., Kirchner, M., Schiemann, J., Roeper, J., Neininger, R., & Schneider,
+G. (2014). A multiple filter test for the detection of rate changes in renewal
+processes with varying variance. The Annals of Applied Statistics, 8(4),2027-2067.
+
+
+Original code
+-------------
+Adapted from the published R implementation:
+DOI: 10.1214/14-AOAS782SUPP;.r
+
+"""
+
+import numpy as np
+import quantities as pq
+
+
+def multiple_filter_test(window_sizes, spiketrain, t_final, alpha, n_surrogates,
+                         test_quantile=None, test_param=None, dt=None):
+    """
+    Detects change points.
+
+    This function returns the detected change points, that correspond to the 
+    maxima of the `_filter_processes`. These are the processes generated by 
+    sliding the windows of step `dt`; at each step the difference between spike
+    on the right and left window is calculated.
+
+    Parameter
+    ---------
+        window_sizes : list of quantity objects
+                    list that contains windows sizes
+        spiketrain : neo.SpikeTrain, numpy array or list
+            spiketrain objects to analyze
+        t_final : quantity
+            final time of the spike train which is to be analysed
+        alpha : integer
+            alpha-quantile for the set of maxima of the limit processes
+        n_surrogates : integer
+            numbers of simulated limit processes
+        dt : quantity
+          resolution, time step at which the windows are slided
+        test_param : np.array 3 * num of window,
+            first row: list of `h`, second and third rows: empirical means and
+            variances of the limit process correspodning to `h`. This will be 
+            used to normalize the `filter_process` in order to give to the every
+            maximum the same impact on the global statistic.
+           
```

Comment:
Explanation for test_quantile is missing, and the order of the arguments is not consistent with that of the definition of the function.